Robust.Client/Graphics/Clyde/Clyde.HLR.cs

```diff
@@ -369,28 +354,39 @@ private void DrawEntities(Viewport viewport, Box2Rotated worldBounds, Box2 world
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private void ProcessSpriteEntities(MapId map, Matrix3 eyeMatrix, Box2Rotated worldBounds,
-            RefList<(SpriteComponent sprite, Matrix3 matrix, Angle worldRot, float yWorldPos)> list)
+        private void ProcessSpriteEntities(MapId map, Viewport view, IEye eye, Box2Rotated worldBounds,
+            RefList<(SpriteComponent sprite, Vector2 worldPos, Angle worldRot, Box2 spriteScreenBB)> list)
         {
+            // Construct a matrix equivalent for Viewport.WorldToLocal()
+
+            eye.GetViewMatrix(out var viewMatrix, view.RenderScale);
+
+            var uiProjmatrix = Matrix3.Identity;
+            uiProjmatrix.R0C0 = EyeManager.PixelsPerMeter;
+            uiProjmatrix.R1C1 = -EyeManager.PixelsPerMeter;
+            uiProjmatrix.R0C2 = view.Size.X / 2f;
+            uiProjmatrix.R1C2 = view.Size.Y / 2f;
+
+            var matrix = viewMatrix * uiProjmatrix;
+
```

Comment:
So there might be much easier ways of doing this.

For one, AFAIK if `DrawEntities` is only ever called from `RenderViewport` like it is atm, then  `_currentMatrixView` should always be equal to the `viewMatrix` here and doesn't need to be calculated?

Also there's probably an easier way to get `uiProjmatrix`? but `EyeManager.GetScreenProjectionMatrix` uses the display-size rather then the viewport size, which I assume matters?

Yeah Im not super sure on this stuff

Reply:
Beats me, probably better to PR to main repo and let pjb have a look when he gets time.

Reply:
Ok, I'll just open a main repo PR.